### DISCUSSION TITLE
class 'Shadow' does not exist

### DISCUSSION THREAD

**Discussion by eimarinin** [tailwindlabs/tailwindcss#13885] [Help]
I have a button with the following style:
`
@apply shadow-[inset_0px_0px_0px_0px_rgba(0, 0, 0, 0)];
`

When compiling a file with this style, it throws an error:
CssSyntaxError: tailwindcss: D:\Web\africa\assets\static\css\input.css:202:13: The `shadow-[inset_0px_0px_0px_0px_rgba(0,` class does not exist. If `shadow-[inset_0px_0px_0px_0px_rgba(0,` is a custom class, make sure it is defined within a `@layer` directive.

Before compiling, I leave this style in <head>, and there are no errors.

I tried a custom class, for example, top-[14px], everything works with it

**Comment by wongjn** [comment]
As the error message states, `shadow-[inset_0px_0px_0px_0px_rgba(0,` would not resolve to a valid Tailwind class.

You may have meant `shadow-[inset_0px_0px_0px_0px_rgba(0,0,0,0)]` or `shadow-[inset_0px_0px_0px_0px_rgba(0,_0,_0,_0)];`. You could also consider not using `@apply` at all:

```css
selector {
  box-shadow: inset 0 0 0 0 rgba(0, 0, 0, 0);
}
```

---

As an aside, Adam Wathan (creator of Tailwind) does seem to advocate avoiding `@apply`:
- https://twitter.com/adamwathan/status/1226511611592085504
- https://twitter.com/adamwathan/status/1559250403547652097
